inc/Engine/Media/AboveTheFold/Frontend/Controller.php

```diff
@@ -437,11 +437,24 @@ private function generate_source_tags( $lcp, $start_tag, $end_tag ) {
 			}
 			// Add the source to the sources array.
 			$sources[] = $source->srcset;
-			// Append the source and media query to the tag string.
-			$tag .= $start_tag . 'href="' . $source->srcset . '" media="' . $media . '"' . $end_tag;
-			// If a max-width is found in the source's media attribute, update the previous max-width.
-			if ( preg_match( '/\(max-width: (\d+(\.\d+)?)px\)/', $source->media, $matches ) ) {
-				$prev_max_width = floatval( $matches[1] );
+
+			if ( ! empty( $source->sizes ) ) {
+				$sizes = 'imagesizes="' . $source->sizes . '"';
+			}
+
+			// Check if the media attribute is empty and the type attribute is not.
+			if ( empty( $media ) && ! empty( $source->type ) ) {
+				// Generate the link tag.
+				$tag .= $start_tag . 'href="' . $source->srcset . '" ' . ( $sizes ?? '' ) . $end_tag;
+				break;
+			} else {
```

Comment:
the usage of `else` here is not necessary if you `break` just before I believe

Reply:
Done